lib/browser.js

```diff
@@ -121,15 +121,26 @@ class Browser {
 
   reconnect (newSocket) {
     if (this.state === EXECUTING_DISCONNECTED) {
+      // In case the socket just lost connection and the browser continued the execution,
+      // we just update the state back to "EXECUTING" so that it matches the current state.
       this.log.debug(`Reconnected on ${newSocket.id}.`)
       this.setState(EXECUTING)
     } else if ([CONNECTED, CONFIGURING, EXECUTING].includes(this.state)) {
+      // In case for some reason the socket has changed, we just bind the events of the
+      // new socket (see below)
       this.log.debug(`New connection ${newSocket.id} (already have ${this.getActiveSocketsIds()})`)
     } else if (this.state === DISCONNECTED) {
-      this.log.info(`Connected on socket ${newSocket.id} with id ${this.id}`)
+      // In case the browser disconnected completely, we want to ensure that the browser will be
+      // registered and starts execution again. This can happen for example if the browser
+      // instance crashed and cannot continue from it's own execution state.
+      this.log.info(`Reconnected on socket ${newSocket.id} with id ${this.id}`)
```

Comment:
How about a better `info` rather than the comment?  EG
`Restarting tests, client returned, now on  ${newSocket.id} with id ${this.id}`

Reply:
That works for me. Done